File: frontend/src/types/notification.ts
export const NotificationType = {
  Reply: 1,
  Like: 2,
  Reputation: 3,
} as const

export type NotificationType = (typeof NotificationType)[keyof typeof NotificationType]

export interface NotificationResponse {
  notificationId: number
  userId: number
  relatedPostId: number | null
  createdAt: string
  type: number
  message: string
  isRead: boolean
}

export interface PagedNotificationsResponse {
  items: NotificationResponse[]
  page: number
  pageSize: number
  totalCount: number
  totalPages: number
  unreadCount: number
}

export interface GetNotificationsParams {
  unreadOnly?: boolean
  page?: number
  pageSize?: number
}

File: UCB-Forum.Server/Models/Notification.cs
namespace UCB_Forum.Server.Models;

public class Notification
{
    public int NotificationId { get; set; }
    public int UserId { get; set; }
    public int? RelatedPostId { get; set; }
    public DateTime CreatedAt { get; set; }
    public byte Type { get; set; }
    public string Message { get; set; } = string.Empty;
    public bool IsRead { get; set; }

    public User User { get; set; } = null!;
    public Post? RelatedPost { get; set; }
}
